micro-deps/micro-deps/src/main/java/com/ofg/infrastructure/discovery/MicroserviceConfiguration.java

```diff
@@ -20,12 +22,11 @@
 
     public MicroserviceConfiguration(ServicePath servicePath, List<Dependency> dependencies) {
         this.servicePath = servicePath;
-        this.dependencies = dependencies;
+        this.dependencies = unmodifiableList(dependencies);
```

Comment:
I'm afraid it's not enough. `unmodifiableList()` returns a read only list and should be used when a view for given list is returned outside. In case of a list passed in a constructor/method call the source list can still be modified and or our "read through" view will reflect performed changes. [Guava](https://code.google.com/p/guava-libraries/wiki/ImmutableCollectionsExplained) probably would be better for that.